src/main/java/org/opensearch/neuralsearch/processor/rerank/ByFieldRerankProcessor.java

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.rerank;
+
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.common.collect.Tuple;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.bytes.BytesReference;
+import org.opensearch.core.xcontent.MediaType;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.neuralsearch.processor.rerank.context.ContextSourceFetcher;
+import org.opensearch.search.SearchHit;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * A reranking processor that reorders search results based on the content of a specified field.
+ * <p>
+ * The ByFieldRerankProcessor extends the RescoringRerankProcessor to provide field-based reranking
+ * capabilities. It allows for reordering of search results by considering the content of a
+ * designated target field within each document.
+ * <p>
+ * Key features:
+ * <ul>
+ *   <li>Reranks search results based on a specified target field</li>
+ *   <li>Optionally removes the target field from the final search results</li>
+ *   <li>Supports nested field structures using dot notation</li>
+ * </ul>
+ * <p>
+ * The processor uses the following configuration parameters:
+ * <ul>
+ *   <li>{@code target_field}: The field to be used for reranking (required)</li>
+ *   <li>{@code remove_target_field}: Whether to remove the target field from the final results (optional, default: false)</li>
+ * </ul>
+ * <p>
+ * Usage example:
+ * <pre>
+ * {
+ *   "rerank": {
+ *     "by_field": {
+ *       "target_field": "document.relevance_score",
```

Comment:
Really like such clear and detailed comments. 

Will the _score changed to the value from target_field after reranking?

Reply:
Yep it updates `_score` value with the target_field, It was suggested to keep the previous score so there is a onboarded `previous_score` field to keep this info

Reply:
Got it, can you explain this in the PR description and this java doc?

Reply:
Thanks for the feedback you can check commit [92e571a](https://github.com/opensearch-project/neural-search/pull/932/commits/92e571a70c3fa1ec494ad57ccd5836036dd02522) 😄